Engine.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <SFML/Graphics.hpp>
+#include "TextureHolder.h"
+#include "Player.h"
+#include "Cupcake.h"
+#include "FrameTimer.h"
+
+
+class Engine
+{
+public:
+	//Constructor
+	Engine();
+
+	//run all the private functions
+	void run();
+	
+
+private:
+	
+	//Private functions. Run will call all private functions 
+	void input();
+	void draw();
+	void updateMousePosition();
+
+	//Prepare the player
+	Player player;
+	//Prepare the cupcake
+	Cupcake cupcake;
+
+	//The usual RenderWindow
+	sf::RenderWindow m_Window;
+
+	//What are the screen height and width? (to be used in spawn() and update())
+	unsigned int m_heightOfScreen = sf::VideoMode::getDesktopMode().height;
+	unsigned int m_widthOfScreen = sf::VideoMode::getDesktopMode().width;
+
+	//Declare a sprite and a texture for the background
+	sf::Sprite m_BackgroundSprite;
+	sf::Texture m_BackgroundTexture;
+
+	//How many seconds, in gametime, have passed?
+	FrameTimer ft;
+	const float m_dt = ft.Mark();
+
+	//Is the game currently playing?
+	bool m_Playing = true;

```

Comment:
This is never `false`. Do you need this member?

Reply:
I planed to keep this as part of the HUD implementation. I still have to figure this part out.